src/components/CartIcon.vue

```diff
@@ -0,0 +1,43 @@
+<template>
+  <div class="cart-icon">
+    <img src="https://cdn.icon-icons.com/icons2/1392/PNG/512/cart_96642.png" alt="Корзина" width="30" height="30"/>
+    <span class="cart-badge">{{ itemQty }}</span>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, PropType } from "vue";
+
+export default defineComponent({
+  name: "CartIcon",
+  props: {
+    itemQty: {
```

Comment:
what is itemQty?

Reply:
may be it is cartCount. However we should have incapsulated cart icon component which contains only picture
Then we should recreate cart-header where there will be no props just cartStore counter.